Agents/AverageSquareMetrePriceAgent/tests/conftest.py

```diff
@@ -0,0 +1,373 @@
+
+        # This will run only once per entire test session
+        # It ensures that the requested PostgreSQL Docker service is ready to accept queries
+        service_available = False
+        while not service_available:
+            try:
+                conn = pg.connect(host='localhost', port=service.host_port,
+                                  user=DB_USER, password=DB_PASSWORD,
+                                  database=DATABASE)
+                if conn.status == pg.extensions.STATUS_READY:
+                    service_available = True
+            except Exception:
+                time.sleep(3)
+
+        return service_url
+    return _get_service_url
+
+
+@pytest.fixture(scope="session")
+def initialise_clients(get_blazegraph_service_url, get_postgres_service_url):
+    # Retrieve "user-facing" endpoints for all clients/services relevant for testing
+    # (i.e. invoked during testing from outside the Docker stack)
+    # --> those shall be `localhost:...` even when agent is running as Docker container
+    
+    # Retrieve endpoint for triple store
+    sparql_endpoint = get_blazegraph_service_url(KG_SERVICE, url_route=KG_ROUTE)
+
+    # Retrieve endpoint for postgres
+    rdb_url = get_postgres_service_url(RDB_SERVICE, url_route=RDB_ROUTE)
+
+    # Create SparqlClient for testing
+    sparql_client = KGClient(sparql_endpoint, sparql_endpoint)
+
+    # Create DerivationClient for creating derivation instances
+    derivation_client = sparql_client.jpsBaseLib_view.DerivationClient(
+        sparql_client.kg_client,
+        DERIVATION_INSTANCE_BASE_URL
+    )
+
+    yield sparql_client, derivation_client, rdb_url
+
+    # Clear logger at the end of the test
+    clear_loggers()
+
+
+# ----------------------------------------------------------------------------------
+# Module-scoped test fixtures
+# (i.e. the fixture is destroyed during teardown of the last test in the module)
+# ----------------------------------------------------------------------------------
+
+@pytest.fixture(scope="module")
+def create_example_agent():
+    def _create_example_agent(
+        register_agent:bool=False,
+        random_agent_iri:bool=False,
+    ):
+        agent_config = config_derivation_agent(AGENT_ENV)
+        agent = AvgSqmPriceAgent(
+            register_agent=agent_config.REGISTER_AGENT if not register_agent else register_agent,
+            agent_iri=agent_config.ONTOAGENT_SERVICE_IRI if not random_agent_iri else 'http://agent_' + str(uuid.uuid4()),
+            time_interval=agent_config.DERIVATION_PERIODIC_TIMESCALE,
+            derivation_instance_base_url=agent_config.DERIVATION_INSTANCE_BASE_URL,
+            kg_url=host_docker_internal_to_localhost(QUERY_ENDPOINT),
+            kg_update_url=host_docker_internal_to_localhost(UPDATE_ENDPOINT),
+            # NOTE For agent endpoint, we keep this as it is for now (i.e. start with http://host.docker.internal)
```

Comment:
@jb2197: This is based on your example agent. I do not fully understand how that properly tests the dockerised agent as the endpoints will always be `localhost` and not use the Docker container internal connections to actually use/test the internal agent connections. 
Your tests and also my tests pass (both locally run as well as dockerised agent ones); however, I have the impression that the tests ar enot fully decoupled from one another. Any clarification and more elaborate description of the design might be very helpful